tools/module/generate.py

```diff
@@ -88,32 +88,61 @@ def generate_cmake(parts):
     )
 
 
-def generate_header(parts):
-    template = textwrap.dedent(
-        """\
-        #ifndef {define}
-        #define {define}
+def generate_header(parts, director):
+    if director:
+        template = textwrap.dedent(
```

Comment:
I'd make these just different constants like "normal_template" and "behaviour_template". Then the context for the if statement